dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ContentReportsRestRepository.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.dspace.app.rest.contentreports.Filter;
+import org.dspace.app.rest.converter.FilteredItemConverter;
+import org.dspace.app.rest.model.ContentReportsSupportRest;
+import org.dspace.app.rest.model.FilteredCollectionRest;
+import org.dspace.app.rest.model.FilteredCollectionsQuery;
+import org.dspace.app.rest.model.FilteredCollectionsRest;
+import org.dspace.app.rest.model.FilteredItemRest;
+import org.dspace.app.rest.model.FilteredItemsQuery;
+import org.dspace.app.rest.model.FilteredItemsQueryPredicate;
+import org.dspace.app.rest.model.FilteredItemsRest;
+import org.dspace.app.rest.projection.Projection;
+import org.dspace.app.rest.utils.FilteredCollectionsReportUtils;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataField;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.contentreports.QueryPredicate;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+/**
+ * This repository serves the content reports ported from DSpace 6.x
+ * (Filtered Collections and Filtered Items).
+ * @author Jean-François Morin (Université Laval)
+ */
+@Component(ContentReportsSupportRest.CATEGORY + "." + ContentReportsSupportRest.NAME)
+public class ContentReportsRestRepository extends AbstractDSpaceRestRepository {
```

Comment:
Again, tests should be added for this RestRepository.  Other RestRepository classes have ITs